### PR TITLE
added instant sort action

### PR DESCRIPTION
**Pull request by mlostekk**
added instant sorting action with the last saved settings or default values

### CODE REVIEW

File: src/org/roana0229/android_xml_sorter/XmlSorterDialog.java [hunk 1]
```diff
@@ -20,12 +20,12 @@
     private static final String TITLE = "Xml Sort Option";
     private static final String OK_BUTTON_TEXT = "Sort";
 
-    private static final String PC_KEY_PREFIX_SPACE_POS = "PC_KEY_PREFIX_SPACE_POS";
-    private static final String PC_KEY_SPACE_BETWEEN_PREFIX = "PC_KEY_SPACE_BETWEEN_PREFIX";
-    private static final String PC_KEY_INSERT_XML_INFO = "PC_KEY_INSERT_XML_INFO";
-    private static final String PC_KEY_DELETE_COMMENT = "PC_KEY_DELETE_COMMENT";
-    private static final String PC_KEY_INPUT_CASE = "PC_KEY_INPUT_CASE";
-    private static final String PC_KEY_CODE_INDENT = "PC_KEY_CODE_INDENT";
+    static final String PC_KEY_PREFIX_SPACE_POS     = "PC_KEY_PREFIX_SPACE_POS";
```
Comment (roana0229) [inline src/org/roana0229/android_xml_sorter/XmlSorterDialog.java]: Please format by default code style. 🙏

File: src/org/roana0229/android_xml_sorter/XmlSorterDialog.java [hunk 2]
```diff
@@ -82,8 +82,14 @@ public boolean isSnakeCase() {
         return mInputCaseBox.getSelectedIndex() == 0;
     }
 
-    public int getPrefixSpacePosition() {
-        return Integer.parseInt(mPrefixSpacePositionBox.getSelectedItem().toString());
+    public int getPrefixSpacePosition()
```
Comment (roana0229) [inline src/org/roana0229/android_xml_sorter/XmlSorterDialog.java]: Please format by default code style. 🙏

**Review by roana0229 — COMMENTED** [review]
Thanks! I will confirm tonight.

**Comment by mlostekk** [comment]
okay, sorry. just check the latest branch commit in feature/instant_sort

File: .idea/misc.xml [hunk 3]
```diff
@@ -38,7 +38,7 @@
       <property name="caretWidth" class="java.lang.Integer" />
     </properties>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" project-jdk-name="IntelliJ IDEA Community Edition IC-172.4343.14" project-jdk-type="IDEA JDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" project-jdk-name="IntelliJ IDEA IU-172.4343.14" project-jdk-type="IDEA JDK">
```
Comment (roana0229) [inline .idea/misc.xml]: Sorry.
Please revert this line, and adapt on only local.
I will add gitignore.

File: src/org/roana0229/android_xml_sorter/XmlSorterAction.java [hunk 4]
```diff
@@ -56,8 +55,25 @@ public void actionPerformed(AnActionEvent event) {
...
 
+    protected void execute(AnActionEvent event,
+                           boolean isSnakeCase,
+                           int prefixSpacePosition,
+                           boolean enableInsertSpaceDiffPrefix,
+                           boolean enableInsertXmlEncoding,
+                           boolean enableDeleteComment,
+                           int codeIndent) {
         // get content
+        final Project project = getEventProject(event);
```
Comment (roana0229) [inline src/org/roana0229/android_xml_sorter/XmlSorterAction.java]: error was displayed at the lower right(! mark), when `meta + L` was used.

```
cannot share data context between Swing events; initial event count = 1763; current event count = 1902
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:132)
	at com.intellij.ide.impl.DataManagerImpl$MyDataContext.getData(DataManagerImpl.java:363)
	at com.intellij.openapi.actionSystem.DataContext.getData(DataContext.java:59)
	at com.intellij.openapi.actionSystem.AnActionEvent.getData(AnActionEvent.java:165)
	at com.intellij.openapi.actionSystem.AnAction.getEventProject(AnAction.java:330)
	at org.roana0229.android_xml_sorter.XmlSorterAction.execute(XmlSorterAction.java:75)
	at org.roana0229.android_xml_sorter.XmlSorterAction.actionPerformed(XmlSorterAction.java:58)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:216)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:233)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher$1.performAction(IdeKeyEventDispatcher.java:576)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.lambda$processAction$2(IdeKeyEventDispatcher.java:625)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:193)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.java:624)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.inInitState(IdeKeyEventDispatcher.java:479)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.dispatchKeyEvent(IdeKeyEventDispatcher.java:213)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:621)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:365)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

Comment (roana0229) [reply src/org/roana0229/android_xml_sorter/XmlSorterAction.java]: error don't display when I changed `execute(AnActionEvent event, ...` to `execute(Project project, Editor editor, ...`.

Can you confirm and change this?

**Review by roana0229 — COMMENTED** [review]
Please check review and fix 🙏

**Comment by mlostekk** [comment]
okay, you are right. Sorry for the issue, now it should be fixed as you suggested.

misc.xml is reverted too.

**Comment by roana0229** [comment]
Thanks, Awesome 👍

**Pull request merged**